src/index.js

```diff
@@ -1,20 +1,51 @@
 /* @flow */
+/* eslint-disable react/no-multi-comp */
 import React from 'react';
 import { NativeModules } from 'react-native';
+import type { Ref } from 'react';
 
 import NativeEventEmitter from './NativeEventEmitter';
 import SubscriberView from './components/SubscriberView';
 import PublisherView from './components/PublisherView';
 
 import type {
   RNOpenTokEventCallback,
-  PublisherViewProps,
-  SubscriberViewProps,
   OpenTokEvent,
+  SubscriberProps,
+  PublisherProps,
 } from './types';
 
 const listeners = {};
 
+export class Subscriber extends React.Component<SubscriberProps> {
+  render() {
+    return <SubscriberView listeners={listeners} {...this.props} />;
+  }
+}
+
+export class Publisher extends React.Component<PublisherProps> {
+  ref: Ref<typeof PublisherView>;
+
+  switchCamera = () => {
+    if (this.ref && typeof this.ref !== 'string') {
```

Comment:
Why we need this check?

Reply:
I'd liked better the previous version – keep the method here, not in actual NativeView. And I also think that currently it's broken on iOS, because iOS side wait for camera={"front" | "back" | "uspecified"} 😢

Reply:
```